src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonic.hpp

```diff
@@ -208,16 +208,18 @@ struct EvolutionMetavars {
             Event,
             tmpl::flatten<tmpl::list<
                 Events::Completion,
-                dg::Events::field_observations<volume_dim, Tags::Time,
-                                               observe_fields,
-                                               analytic_solution_fields>,
+                dg::Events::field_observations<
+                    volume_dim, Tags::Time, observe_fields,
+                    tmpl::conditional_t<
+                        evolution::is_numeric_initial_data_v<initial_data>,
+                        tmpl::list<>, analytic_solution_fields>>,
```

Comment:
One question on the design:

Can we observe constraint norms also in the case where we _have_ an analytic solution? Perhaps it would be appropriate to pass two lists of tags to the event, instead of one list and a bool?

Reply:
The current design can output constraint norms when there is an analytic solution in the databox, but it can't output both constraint norms and analytic solution norms at the same time.

Making both print at the same time would need a different design, because the tensors currently are a parameter pack. How would you tell where one pack begins and the other ends?

I think the capability might be nice, but it is marginally important. Usually you only care about constraint norms when there is no analytic solution to use for your error norms. So I'd prefer to stick with this design for now, to avoid getting bogged down on this when there are more pressing things to do to get BBHs going. We could always revisit this if/when it turns out there's a need for simultaneous output of analytic solution norms and constraint norms.